### DISCUSSION TITLE
Backup script example

### DISCUSSION THREAD

**Discussion by cod3rshotout** [evertramos/server-automation#2] [Q&A]
Hi, I would like to use the `server-automation` to automatize the database backup of all my containers (wordpress and other apps).

I have cloned the repository and then within the cloned directory I changed the `.env` file setting the home path.
When I execute this command: `./backup.sh` I get:

```
 Start execution '/home/john/docker/server-automation/backup/backup.sh ':
 ✔ success Function 'check_docker'
 ✔ success Function 'check_running_script'
 ✔ success Function 'starts_initial_check'
 ✔ success Function 'check_server_automation_env_file_exists'
 ✘ ERROR   Folder '/home/john/server-automation/sites' does not exist
```

I also noticed that the youtube video is outdate and the wiki link point to a 404.
Could you please help me to configure this?

**Comment by evertramos** [comment]
Hello @cod3rshotout,

Add `--source=` and `--destination=` to your script execution and you should be fine.

Let me know if you got it working.

**Comment by cod3rshotout** [comment]
@evertramos Thanks for the tip, after two question I get:

? question Do you want to send to backup storage? 
   (default: NO) [y/N]y

 ✔ success Function 'confirm_user_action'
 ? question Do you want to DELETE the local file after sending this file to backup storage? 
   (default: NO) [y/N]y

 ✔ success Function 'confirm_user_action'
 ✔ success Function 'backup_compress_folder'
 ✔ success Function 'backup_check_tar_file_integrity'
 𝝯 INFO Uploading backup file to Storage ()
 ✘ ERROR   You must inform the required argument(s) to the function: 'ftp_send_file_ftp'

what I need to do?

Thanks

**Comment by evertramos** [comment]
nice!

You need to configure the sftp server here:

https://github.com/evertramos/server-automation/blob/22fcb94c03fb8c253c5f218f6bc0e326c718068d/.env.example#L80

I would suggest try backup locally and then send it to ftp server.

**Comment by cod3rshotout** [comment]
> nice!
> 
> You need to configure the sftp server here:
> 
> https://github.com/evertramos/server-automation/blob/22fcb94c03fb8c253c5f218f6bc0e326c718068d/.env.example#L80
> 
> I would suggest try backup locally and then send it to ftp server.

How can I test the backup locally? Could you show an example please?

**Comment by evertramos** [comment]
Sure!

## Using [Easy-Server](https://github.com/evertramos/easy-server) and [NGINX-proxy-automation](https://github.com/evertramos/nginx-proxy-automation)

In the helper `./backup.sh --help` you will find how to use it. If you are running the nginx-proxy-automation and has your sites at ~/server/sites, you could run the following:

```bash
cd /server-automation/backup
./backup.sh --yes
```

Then your backup files would be in /server/backup

## Server-Automation only 

If you are running server-automation in your own setup you would run:

```bash
cd /server-automation/backup
mkdir -p ~/backup
./backup.sh -s /path_where_your_sites_are -d ~/backup --all-sites --yes
```

This will backup all folders, inside ´/path_where_your_sites_are` at `~/backup`, example:

```bash
/path_where_your_sites_are
  --/site1
  --/site2
  --/site3

~/backup
  --/site1-20220308_00000.tar.gz
  --/site2-20220308_00000.tar.gz
  --/site3-20220308_00000.tar.gz
```

hope it helps!

**Reply by evertramos** [reply]
@cod3rshotout did you manage to use it?

I wonder if we might set a fullpath for final backup, which would instead of getting all folders (we will keep this feature as well) we tar the informed folder, what do you think?
